File: server/app/schemas/task.py
"""Pydantic schemas for Task responses."""

import datetime
import uuid
from pydantic import BaseModel


class TaskFileResponse(BaseModel):
    model_config = {"from_attributes": True}

    id: uuid.UUID
    filename: str
    file_size: int | None
    is_primary: bool
    sort_order: int


class TaskResponse(BaseModel):
    model_config = {"from_attributes": True}

    id: uuid.UUID
    filename: str
    file_size: int | None
    status: str
    current_step: str | None
    progress: int
    error_message: str | None
    extracted_data: dict | None = None
    files: list[TaskFileResponse] = []
    created_at: datetime.datetime
    started_at: datetime.datetime | None
    completed_at: datetime.datetime | None


class TaskListResponse(BaseModel):
    items: list[TaskResponse]
    total: int
    page: int
    page_size: int

File: web/src/types/task.ts
export interface TaskFile {
  id: string
  filename: string
  file_size: number | null
  is_primary: boolean
  sort_order: number
}

export interface Task {
  id: string
  filename: string
  file_size: number | null
  status: string
  current_step: string | null
  progress: number
  error_message: string | null
  extracted_data: Record<string, any> | null
  created_at: string
  started_at: string | null
  completed_at: string | null
  files?: TaskFile[]
}

export interface ProgressEvent {
  step: string
  detail?: string
  progress: number
  current_module?: string
  modules_done?: number
  modules_total?: number
  error?: string
}
